### DISCUSSION TITLE
Invalid event for type=ref,event=workflow_dispatch

### DISCUSSION THREAD

**Discussion by Floppy** [docker/metadata-action#378] [Q&A] [answered]
I'm trying to get a manual trigger for a docker build going (mainly for testing the build when I want to). I've set up a `type=ref` metadata tag, which I want to be used when it gets a `workflow_dispatch` action. However, I get the error message in the title.

The docs at https://github.com/docker/metadata-action?tab=readme-ov-file#typeref list the events available as `branch`, `pr`, and `push`, but the table below includes `workflow_dispatch` as well, which I think has led me to believe that it's an option, whereas in fact I think it might just be a ref example?

So, my question is, should I be able to use `workflow_dispatch` events with `type=ref`, or should I be using a different type? If so, where _is_ `workflow_dispatch` supported?

**Comment by crazy-max** [comment, marked as answer]
> So, my question is, should I be able to use `workflow_dispatch` events with `type=ref`,

`workflow_dispatch` is a github event, not a ref type. As stated in the docs, only `branch`, `tag` and `pr` are valid ref type: https://github.com/docker/metadata-action?tab=readme-ov-file#typeref

> the table below includes `workflow_dispatch` as well, which I think has led me to believe that it's an option, whereas in fact I think it might just be a ref example?

It's just an event type triggering the workflow.

___

`workflow_dispatch` only handles branch ref so in your case it would only work with `type=ref,event=branch` in `tags`. More info: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch

Edit: Ah tags as well so `type=ref,event=tag` should work:

![image](https://github.com/docker/metadata-action/assets/1951866/74508a20-cfdd-49b1-9b4d-1e20d7ee94a7)

**Comment by Floppy** [comment]
Thanks, I've been getting my ref types and events mixed up for one thing. So, `type=ref` can't be used with a `workflow_dispatch` event. I'm having a go at using `type=raw` instead, that seems to be working at the moment. I'll report back with my eventual solution for future reference.

I think that table below the list of supported events is a bit confusing, I went straight to reading that rather than the actual list above it.

**Comment by Floppy** [comment]
Yep, that did it, thanks for the help! I'm using `type=raw` and passing the tag name from an input on the `workflow_dispatch`. For anyone wondering this in future, here's my code:

https://github.com/manyfold3d/manyfold/blob/ab836e3c22f62423b91424cf2340cfde0409bbde/.github/workflows/docker.yml#L34
